prismic-configuration.js

```diff
@@ -1,28 +1,45 @@
-module.exports = {
-  // -- Prismic API endpoint
-  // Determines which repository to query and fetch data from
-  // Configure your site's access point here
-  apiEndpoint: 'https://your-repo-name.prismic.io/api/v2',
+import Prismic from 'prismic-javascript'
+import Link from 'next/link'
 
-  // -- Access Token if the repository is not public
-  // Generate a token in your dashboard and configure it here if your repository is private
-  // accessToken: '########',
+// -- Prismic API endpoint
+// Determines which repository to query and fetch data from
+// Configure your site's access point here
+export const apiEndpoint = 'https://your-repo-name.prismic.io/api/v2'
 
-  // -- Link resolution rules
-  // Manages links to internal Prismic documents
-  // Modify as your project grows to handle any new routes you've made
-  linkResolver: function (doc) {
-    if (doc.type === 'post') {
-      return `/blog/${doc.uid}`
-    }
-    return '/'
-  },
+// -- Access Token if the repository is not public
+// Generate a token in your dashboard and configure it here if your repository is private
+export const accessToken = ''
 
-  // Additional helper function for Next/Link components
-  hrefResolver: function (doc) {
-    if (doc.type === 'post') {
-      return `/post?uid=${doc.uid}`
-    }
-    return '/'
+// -- Link resolution rules
+// Manages links to internal Prismic documents
+// Modify as your project grows to handle any new routes you've made
+export const linkResolver = (doc) => {
+  if (doc.type === 'post') {
+    return `/blog/${doc.uid}`
   }
+  return '/'
 }
+
+// Additional helper function for Next/Link components
+export const hrefResolver = (doc) => {
+  if (doc.type === 'post') {
+    return `/post?uid=${doc.uid}`
+  }
+  return '/'
+}
+
+let frontClient
+
+export const Client = (req = null) => {
+  if (!req && frontClient) return frontClient // prevent generate new instance for client side since we don't need the refreshed request object.
+  else {
+    const options = Object.assign({}, req ? { req } : {}, accessToken ? { accessToken: accessToken } : {})
+    return Prismic.client(apiEndpoint, options)
+  }
+}
+
+export const customLink = (type, element, content, children, index) => (
```

Comment:
`customLink` has been removed for this version since serializeHyperlink hasn't been updated yet to deal with all the cases `htmlSerializer` handles, and both can't be used at the same time. Will return once prismic-reactjs is updated